navigation/navigation-common/src/jbMain/kotlin/androidx/navigation/NavGraph.jb.kt

```diff
@@ -79,17 +91,70 @@ public actual open class NavGraph actual constructor(
         }
     }
 
+    /**
+     * Finds a destination in the collection by ID. This will recursively check the [parent][parent]
+     * of this navigation graph if node is not found in this navigation graph.
+     *
+     * @param resId ID to locate
+     * @return the node with ID resId
+     */
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    public fun findNode(resId: Int): NavDestination? =
+        findNodeComprehensive(resId, this, false)
```

Comment:
This is breaking `setStartDestination`. It used to search the children, which is what `setStartDestination` needs to do, but it is now searching the parents, and so will never find the correct destination, throwing:

```
Cannot find startDestination com.eygraber.vice.samples.nav.shared.Routes.Settings.SettingsHome from NavGraph. Ensure the starting NavDestination was added with route from KClass.
Error
    at kotlin.captureStackTrace (nav-sample-wasm.uninstantiated.mjs:17:1)
    at <com.eygraber:wasmJsApp>.kotlin.captureStackTrace__externalAdapter (nav-sample-wasm.wasm:0x38e398)
    at <com.eygraber:wasmJsApp>.kotlin.Throwable.<init> (nav-sample-wasm.wasm:0x38e1b1)
    at <com.eygraber:wasmJsApp>.kotlin.Throwable.<init> (nav-sample-wasm.wasm:0x38e227)
    at <com.eygraber:wasmJsApp>.kotlin.Exception.<init> (nav-sample-wasm.wasm:0x3a818e)
    at <com.eygraber:wasmJsApp>.kotlin.RuntimeException.<init> (nav-sample-wasm.wasm:0x3a7fa5)
    at <com.eygraber:wasmJsApp>.kotlin.IllegalStateException.<init> (nav-sample-wasm.wasm:0x3a8282)
    at <com.eygraber:wasmJsApp>.androidx.navigation.NavGraph.setStartDestination (nav-sample-wasm.wasm:0x61fd70)
    at <com.eygraber:wasmJsApp>.androidx.navigation.NavGraph.setStartDestination (NavGraph.jb.kt:256:9)
    at <com.eygraber:wasmJsApp>.androidx.navigation.NavGraphBuilder.build (nav-sample-wasm.wasm:0x620771)
```

However, I only get this error on wasmJs and not jvm, which is odd because I would think that it should break both of them.

Reply:
:thinking: maybe a red herring, because flipping that to `true` doesn't fix the problem.

Reply:
Looks like it might be the collection issue on wasm. I put some logging in and even though the id is in nodes, it still returns null

```
NavDestinationBuilder for SettingsHome is 794692654
setStartDestination for SettingsHome is 794692654
Nodes are {794692654=Destination route=com.eygraber.vice.samples.nav.shared.Routes.Settings.SettingsHome, 1622916055=Destination route=com.eygraber.vice.samples.nav.shared.Routes.Settings.AboutUs}
Looking up nodes[794692654] results in null
```

Reply:
It's from https://github.com/androidx/androidx/commit/25b447c7e318e81c9c150e40337e544626b81de9

> but it is now searching the parents

Nope https://github.com/JetBrains/compose-multiplatform-core/blob/dc1870e6aeb5082b5d1fe7b806097586d896597f/navigation/navigation-common/src/jbMain/kotlin/androidx/navigation/NavGraph.jb.kt#L116

Reply:
Yeah, sorry I was reading it wrong. It looks like there's still something wrong with `SparseArrayCompat` on wasmJs though. Debugging that now.

Reply:
OK I figured it out, and then the solution looked very familiar to me. Then I remembered that you sent it to me on Slack last week :sweat_smile: https://github.com/JetBrains/compose-multiplatform-core/pull/1401/commits/0ad8d03113bc432332c5b2163c5091917a5831ef

Reply:
```kotlin
@Suppress("NOTHING_TO_INLINE")
internal fun <E> SparseArrayCompat<E>.commonGet(key: Int): E? {
    return internalGet<E, E?>(key, null)
}
```
was the only patching needed to get my code to work.

```kotlin
@Suppress("NOTHING_TO_INLINE")
internal fun <E> SparseArrayCompat<E>.commonGet(key: Int, defaultValue: E): E {
    return internalGet<E, E>(key, defaultValue)
}
```
shows that `<E, E>` doesn't need to be explicit, and a quick test shows that it works without `<E, E>`.

`SimpleArrayMap` doesn't seem to be used here so it isn't blocking anything.